File: front-end/src/app/endpoints/cv-endpoint/get-by-id/cv-get-by-id-response.ts
export interface CVGetByIdResponse {
  id: number
  kandidatId: string,
  naziv: string,
  objavljen: boolean,
  ime: string;
  prezime: string;
  email:string;
  brojTelefona: string | null;
  drzava: string | null;
  grad: string | null;
  datumModificiranja: Date,
  profesionalniSazetak: string | null;
  vjestine:  {
    $values: string[]
  }
  tehnickeVjestine: {
    $values: string[]
  }
  kursevi: {
    $values: string[]
  }
  edukacija:  {
   $values: EdukacijaResponse[]
}
  zaposlenje:  {
    $values: ZaposlenjeResponse[]
  }
  url:  {
    $values: UrlResponse[]
  }
}

export interface EdukacijaResponse{
  id: number;
  nazivSkole: string;
  datumPocetka: Date | null;
  datumZavrsetka: Date | null;
  grad: string | null;
  opis: string | null;
}

export interface ZaposlenjeResponse {
  id: number;
  nazivKompanije: string;
  nazivPozicije: string;
  datumPocetka: string;
  datumZavrsetka: string | null;
  opis: string | null;
}

export interface UrlResponse {
  id: number;
  naziv: string;
  putanja: string;
}

File: JobSearchingWebApp/Endpoints/CV/GetById/CVGetByIdResponse.cs
using JobSearchingWebApp.Models;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace JobSearchingWebApp.Endpoints.CV.GetById
{
    public class CVGetByIdResponse
    {
        public int Id { get; set; }

        public string KandidatId { get; set; }

        public bool Objavljen { get; set; }

        public string Naziv { get; set; }

        public string Ime { get; set; }

        public string Prezime { get; set; }

        public string Email { get; set; }

        public string? PhoneNumber { get; set; }

        public string? Drzava { get; set; }

        public string? Grad { get; set; }

        public string? ProfesionalniSazetak { get; set; }

        public DateTime DatumModificiranja { get; set; }

        public List<string>? Vjestine { get; set; }

        public List<string>? TehnickeVjestine { get; set; }

        public List<string>? Kursevi { get; set; }

        public List<EdukacijaResponse>? Edukacija { get; set; }

        public List<ZaposlenjeResponse>? Zaposlenje { get; set; }

        public List<URLResponse>? URL { get; set; }
    }

    public class EdukacijaResponse
    {
        public int Id { get; set; }

        public string NazivSkole { get; set; }

        public DateOnly? DatumPocetka { get; set; }

        public DateOnly? DatumZavrsetka { get; set; }

        public string? Grad { get; set; }

        public string? Opis { get; set; }
    }

    public class ZaposlenjeResponse
    {
        public int Id { get; set; }

        public string NazivKompanije { get; set; }

        public string NazivPozicije { get; set; }

        public DateOnly DatumPocetka { get; set; }

        public DateOnly? DatumZavrsetka { get; set; }

        public string? Opis { get; set; }
    }

    public class URLResponse
    {
        public int Id { get; set; }

        public string Naziv { get; set; }

        public string Putanja { get; set; }
    }
}
